CMakeLists.txt

```diff
@@ -31,29 +23,26 @@ endif ()
 
 if ( WIN32 )
    set( BUILD_TESTS False CACHE BOOL "build tests"  )
-else ()
-   set( BUILD_TESTS True CACHE BOOL "build tests"  )
-endif ()
-
-set( BUILD_SHARED_LIBS True CACHE BOOL "build as shared library"  )
+endif()
 
-option( BUILD_TESTS "Build test programs" ON)
+# ObjCrypto compiler options
 option( OBJ_CRYPTO_USE_BORINGSSL "Use BoringSSL for crypto" )
-option( BUILD_SHARED_LIBS "Build as shared library" )
-
-
-set( CMAKE_CXX_STANDARD 17 )
-set( CMAKE_C_STANDARD 17 )
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_C_STANDARD_REQUIRED ON)
+option( OBJ_CRYPTO_OSX "Build for OSX Architectures")
+option( OBJ_CRYPTO_BUILD_TESTS "Build test programs" ON )
+option( OBJ_CRYPTO_BUILD_SHARED "Build as dynamic library" ON )
+option( OBJ_CRYPTO_INSTALL "Install library to system" ON )
 
-message( "OBJ_CRYPTO_USE_BORINGSSL is set to ${OBJ_CRYPTO_USE_BORINGSSL} " )
-message( "BUILD_TESTS  is set to ${BUILD_TESTS} " ) 
-message( "BUILD_SHARED_LIBS is set to ${BUILD_SHARED_LIBS} " ) 
-message( "CMAKE_OSX_ARCHITECTURES is set to ${CMAKE_OSX_ARCHITECTURES} " )
+message( "OBJ_CRYPTO_USE_BORINGSSL is set to ${OBJ_CRYPTO_USE_BORINGSSL}" )
+message( "OBJ_CRYPTO_OSX is set to ${OBJ_CRYPTO_OSX}" )
+message( "OBJ_CRYPTO_BUILD_TESTS  is set to ${OBJ_CRYPTO_BUILD_TESTS}" ) 
+message( "OBJ_CRYPTO_BUILD_SHARED is set to ${OBJ_CRYPTO_BUILD_SHARED}" ) 
+message( "OBJ_CRYPTO_INSTALL is set to ${OBJ_CRYPTO_INSTALL}" )
 
-include(CTest)
+set(BUILD_SHARED_LIBS ${OBJ_CRYPTO_BUILD_SHARED})
 
+if ( OBJ_CRYPTO_OSX)
+    set( CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "OSX archs")
```

Comment:
I'm thinking we should take this out of the CAMKE and it can just be set on CLI to cmake

Reply:
Fixed.